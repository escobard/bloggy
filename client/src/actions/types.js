export const ADD_QUESTION = "ADD_QUESTION",
	ADD_USER = "ADD_USER";
export const UPDATE_USER = "UPDATE_USER";
export const SELECT_QUESTION = "SELECT_QUESTION";
export const FETCH_QUESTIONS = "FETCH_QUESTIONS";
export const UPDATE_QUESTION = "UPDATE_QUESTION";
