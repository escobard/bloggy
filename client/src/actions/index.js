import axios from "axios";

import { server } from "../constants/config";
import { FETCH_USER } from "./types";

export const fetchUser = () => {
	// this fetches our user data, which is only present in the API if the user has been authenticated
	// with google, handled by the passport google strategy

	// as a refreshed with redux thunk, the middleware looks for any returned FUNCTIONS
	// within an action creator, handling these first
	return function(dispatch) {
		axios
			.get("/api/current_user")

			// then returns the data via an action creator object
			.then(res => dispatch({ type: FETCH_USER, payload: res }));
	};
};
