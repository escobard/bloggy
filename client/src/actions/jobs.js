import {SUBMIT_JOB} from './types'

export const submitJob = values =>{
	return {
		type: SUBMIT_JOB,
		payload: values
	}
}