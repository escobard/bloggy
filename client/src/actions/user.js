import axios from "axios";

import { FETCH_USER, LOGOUT_USER } from "./types";

export const fetchUser = () =>
	// this fetches our user data, which is only present in the API if the user has been authenticated
	// with google, handled by the passport google strategy

	// as a refreshed with redux thunk, the middleware looks for any returned FUNCTIONS
	// within an action creator, handling these first
	/*
		return function(dispatch) {
		axios
			.get(`${server}/api/current_user`)

			// then returns the data via an action creator object
			.then(res => dispatch({ type: FETCH_USER, payload: res }));
	};
	*/
	async dispatch => {
		// grabs the user profile object from the API, attaches it to res.data
		// could be refactored into a `routes constant` although largely unecessary
		// until the application scales further
		const res = await axios.get("/api/current_user");

		// console.log("RES", res.data);
		// then returns the data via an action creator object
		dispatch({ type: FETCH_USER, payload: res.data });
	};

export const logoutUser = () => async dispatch => {
	// triggers the logout function within the API with the a get request to the logout route
	const res = await axios.get("/api/logout");
	dispatch({ type: LOGOUT_USER, payload: "" });
};

// to handle our stripe payment token
export const handleToken = token => async dispatch => {
	// passes the token to our route, assuming we have a route in the API
	// route will be created shortly
	const res = await axios.post("/api/stripe", token);

	// since we are effectively just updating the user data, we can use the same action type
	// as FETCH_USER - this is most likely not best practice and is a simple shortcut in this scenario
	dispatch({ type: FETCH_USER, payload: res.data });
};
