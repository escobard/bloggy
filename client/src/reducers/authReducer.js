import { FETCH_USER } from "../actions/types";

export default function(state = null, action) {
	let { type, payload } = action;
	switch (type) {
		case FETCH_USER:
			// console.log(action);

			// essentiall this checks if the returned value is an empty string
			// an empty string is returned when the user is not authenticated by google
			// this allows us to test for cases when the user is not authenticated in, by checking the
			// value of payload
			return payload || false;
		default:
			return state;
	}
}
