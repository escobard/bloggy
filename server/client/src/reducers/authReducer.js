import { FETCH_USER } from "../actions/types";

export default function(state = {}, action) {
	let { type, payload } = action;
	switch (type) {
		case FETCH_USER:
			console.log(action);
			return state;
		default:
			return state;
	}
}
