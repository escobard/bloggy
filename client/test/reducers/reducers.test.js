import authReducer from "../../src/reducers";
import { FETCH_USER, LOGOUT_USER } from "../../src/actions/types";
import toJson from "enzyme-to-json";

describe("Reducers", () => {
	it("renders the authReducer, with no state", () => {
		expect(authReducer(undefined, {})).toEqual({ auth: null });
	});
	it("renders the authReducer, with un authenticated response", () => {
		expect(
			authReducer({ auth: "" }, { type: FETCH_USER, payload: "" })
		).toEqual({
			auth: false
		});
	});
	it("renders the authReducer, with authenticated response", () => {
		expect(
			authReducer({ auth: {} }, { type: FETCH_USER, payload: {} })
		).toEqual({
			auth: {}
		});
	});
	it("matches authReducer snapshot", () => {
		expect(toJson(authReducer(undefined, {}))).toMatchSnapshot();
	});
});
