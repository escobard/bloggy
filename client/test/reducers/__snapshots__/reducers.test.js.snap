// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducers matches authReducer snapshot 1`] = `undefined`;
